src/main/java/com/google/sps/servlets/RetrieveActivitiesServlet.java

```diff
@@ -0,0 +1,105 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+
+import com.google.sps.data.Activity;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+
+/** 
+* Servlet that retrieves actvities of a specified category.
+*/
+@WebServlet("/retrieve-activities")
+public class RetrieveActivitiesServlet extends HttpServlet {
+  private static final String GAMES_CATEGORY = "games";
+  private static final String VIDEOS_CATEGORY = "videos";
+  private static final String ARTICLES_CATEGORY = "articles";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String activityType = getActivityType(request);
+    List<Activity> activities;
+
+    switch(activityType) {
+      case GAMES_CATEGORY:
+        activities = getGames();
+        break;
+      case VIDEOS_CATEGORY:
+        activities = getVideos();
+        break;
+      case ARTICLES_CATEGORY:
+        activities = getArticles();
+        break;
+      default:
+        activities = null;
```

Comment:
Is this a valid case? If not, consider adding some error message. Simply printing some error message to the log tool is fine (since this is an internal error).

Reply:
Ok

Reply:
Fixed it.